cogs/welcome/welcome.py

```diff
@@ -132,6 +199,139 @@ async def logServerLeave(self, leaveUser: discord.Member):
+                )
+            )
+            try:
+                response = await self.bot.wait_for("message", timeout=30.0, check=check)
+            except asyncio.TimeoutError:
+                await ctx.send("You took too long, not overwriting")
+                return
+
+            if response.content.lower() != "yes":
+                await ctx.send("Not overwriting the greeting")
+                return
+
+        # save the greetings
+        greetings[name] = greeting.content
+        await greeting.add_reaction("✅")
+        await self.config.guild(ctx.guild).greetings.set(greetings)
+        return
+
+    @checks.mod_or_permissions()
+    @welcome.command(name="remove", aliases=["delete", "del", "rm"])
+    async def greetRemove(self, ctx: Context, name: str):
+        """
+        Removes a greeting.
+
+        Parameters:
+        -----------
+        name: name of the greeting to remove
+        """
+
+        def check(message: discord.Message):
+            return message.author == ctx.message.author and message.channel == ctx.message.channel
+
+        greetings = await self.config.guild(ctx.guild).greetings()
+        if name in greetings:
+            await ctx.send(
+                warning("Are you sure you wish to delete this greeting? Respond with 'yes' if yes")
+            )
+            try:
+                response = await self.bot.wait_for("message", timeout=30.0, check=check)
+            except asyncio.TimeoutError:
+                await ctx.send("You took too long, not deleting")
+                return
+
+            if response.content.lower() != "yes":
+                await ctx.send("Not deleting")
+                return
+
+            # delete the greeting
+            del greetings[name]
+            await ctx.send(f"{name} removed from list")
+            await self.config.guild(ctx.guild).greetings.set(greetings)
+        else:
+            ctx.send(f"{name} not in list of greetings")
+        return
+
+    @checks.mod_or_permissions()
+    @welcome.command(name="list", aliases=["ls"])
+    async def greetList(self, ctx: Context):
+        """
+        List's all the greetings on the server
+        """
+        greetings = await self.config.guild(ctx.guild).greetings()
+
+        if not greetings:
+            await ctx.send("There are no greetings, please add some first!")
+            return
+
+        msg = ""
+
+        for key in greetings:
+            msg += f"{key}: {greetings[key]}\n"
+
+        pageList = []
+        pages = list(pagify(msg, page_length=500))
+        totalPages = len(pages)
+        async for pageNumber, page in AsyncIter(pages).enumerate(start=1):
+            embed = discord.Embed(
+                title=f"Server greetings changes for {ctx.guild.name}", description=page
+            )
+            embed.set_footer(text=f"Page {pageNumber}/{totalPages}")
+            pageList.append(embed)
+        await menu(ctx, pageList, DEFAULT_CONTROLS)
```

Comment:
Curious, how does this look with multiple long greetings?

Reply:
The greetings have multiple pages then it appears.

Reply:
Ah okay, we can leave it for now, and change it later if we think this should be expanded.